stmocli/cli.py

```diff
@@ -23,26 +24,67 @@ def init(conf):
 @cli.command()
 @pass_conf
 @click.argument('query_id')
-@click.argument('query_name')
+@click.argument('file_name')
 @click.option(
     '--redash_api_key',
     default=lambda: os.environ.get('REDASH_API_KEY', '')
 )
-def track(conf, query_id, query_name, redash_api_key):
+def track(conf, query_id, file_name, redash_api_key):
     # Get query: https://github.com/getredash/redash/blob/1573e06e710733714d47940cc1cb196b8116f670/redash/handlers/api.py#L74
     redash = RedashClient(redash_api_key)
     url_path = 'queries/{}?api_key={}'.format(query_id, redash_api_key)
-    results, response = redash._make_request(
-        requests.get, 
-        urljoin(redash.BASE_URL, url_path)
-    )
-    query = results['query']
+    try:
+        results, response = redash._make_request(
+            requests.get,
+            urljoin(redash.API_BASE_URL, url_path)
+        )
 
-    filename = query_name + '.sql'
-    with open(filename, 'w') as outfile:
-        outfile.write(query)
+        query = results['query']
+        with open(file_name, 'w') as outfile:
+            outfile.write(query)
 
-    conf.add_query(query_name, query_id, filename)
+        query_meta = {
+            "id": query_id,
+            "data_source_id": results['data_source_id'],
+            "name": results['name'],
+            "description": results['description'],
+            "schedule": results['schedule'],
+            "options": results['options']
+        }
+        conf.add_query(file_name, query_meta)
+        click.echo("Tracking Query ID {} in {}".format(query_id, file_name))
+    except RedashClient.RedashClientException as e:
+        click.echo("Failed to track Query ID {}: {}".format(query_id, e))
+
+
+@cli.command()
+@pass_conf
+@click.argument('file_name')
+@click.option(
+    '--redash_api_key',
+    default=lambda: os.environ.get('REDASH_API_KEY', '')
+)
+def push(conf, file_name, redash_api_key):
+    redash = RedashClient(redash_api_key)
+
+    try:
+        meta = conf.get_query(file_name)
+        with open(file_name, 'r') as fin:
+            query = fin.read()
+
+        try:
+            redash.update_query(meta['id'], meta['name'],
+                                query, meta['data_source_id'],
+                                meta['description'], meta['options'])
+            m = md5.new()
```

Comment:
Why include MD5? Testability?

Reply:
Yup, I was having trouble coming up with meaningful tests, this gives an easily double-checkable way to make sure the correct contents were pushed.